common/testing/taskpoller/taskpoller.go

```diff
@@ -84,70 +96,139 @@ func New(
 	}
 }
 
-// PollWorkflowTask issues PollWorkflowTaskQueueRequests to obtain a new workflow task.
+// PollWorkflowTask creates a workflow task poller that uses the given PollWorkflowTaskQueueRequest.
 func (p *TaskPoller) PollWorkflowTask(
-	tv *testvars.TestVars,
-	funcs ...OptionFunc,
-) (*workflowservice.PollWorkflowTaskQueueResponse, error) {
-	p.t.Helper()
-	options := newOptions(tv, funcs)
-	ctx, cancel := newContext(options)
-	defer cancel()
-	return p.pollWorkflowTask(ctx, options)
+	req *workflowservice.PollWorkflowTaskQueueRequest,
+) *workflowTaskPoller {
+	return &workflowTaskPoller{TaskPoller: p, pollWorkflowTaskRequest: req}
 }
 
-// PollAndHandleWorkflowTask issues PollWorkflowTaskQueueRequests to obtain a new workflow task,
+// PollAndHandleWorkflowTask issues a PollWorkflowTaskQueueRequest to obtain a new workflow task,
 // invokes the handler with the task, and completes/fails the task accordingly.
+// Any unspecified but required request and response fields are automatically generated using `tv`.
+// Returning an error from `handler` fails the task.
+// If no task is available, it returns NoTaskAvailable.
 func (p *TaskPoller) PollAndHandleWorkflowTask(
 	tv *testvars.TestVars,
 	handler func(task *workflowservice.PollWorkflowTaskQueueResponse) (*workflowservice.RespondWorkflowTaskCompletedRequest, error),
-	funcs ...OptionFunc,
+	funcs ...optionFunc,
```

Comment:
This should be 
```suggestion
	opts ...optionFunc,
```
everywhere.